src/interp.rs

```diff
@@ -19,153 +19,483 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-use smallvec::SmallVec;
-use std::f64;
-
-/// Data structure for polyphase FIR interpolator
+#[allow(clippy::large_enum_variant)]
 #[derive(Debug)]
-pub struct Interp {
-    /// Interpolation factor of the interpolator
-    factor: usize,
-    /// Taps (prefer odd to increase zero coeffs)
-    taps: usize,
-    /// Number of channels
-    channels: u32,
-    /// Size of delay buffer
-    delay: usize,
-    /// List of subfilters (one for each factor)
-    filter: SmallVec<[Filter; 4]>,
-    /// List of delay buffers (one for each channel)
-    z: Box<[f32]>,
-    /// Current delay buffer index
-    zi: usize,
+enum _Interp {
```

Comment:
```suggestion
enum InterpInner {
```

Seems nicer than prepending a `_`